### ISSUE TITLE
External state storage calls mutator when not needed

### ISSUE THREAD

**Issue by jukkasi** [stateless4j/stateless4j#37]
I have an issue where the state is maintained in database. Constructor "Construct a state machine with external state storage" triggers database update in construction phase calling doIt for mutator. 

However this creates extra updates every time state machine is created since I have just loaded the state from database, giving it as an ininitalState and doing an update right after this is not necessary. Also I'm bit confused why initialState is given separate and not fetched using given accessor.

Any ideas how the constructor should really work in this case? I would like to avoid unnecessary database calls but let state machine still do the updates.

**Comment by cwash** [comment]
I had a similar question as to how this should be used.  I noticed that the mutator is called by the StateMachine external storage constructor, which also takes the initialState.  But the problem is we need to know what that the initialState is before we call the constructor.  So I have to perform the following in order to get things to work properly:

        StateMachineConfig<State, Trigger> stateMachineConfig = new StateMachineConfig<>();

        // config

        MongoPersistorMutator<State> mutator = new MongoPersistorMutator<>();
        MongoPersistorAccessor<State> accessor = new MongoPersistorAccessor<>();

        State lastState = accessor.call();
        navigationMachine = new StateMachine<>(lastState, accessor, mutator, stateMachineConfig);

Things seem to work properly, I notice the mutator being called after fire and transitions a few times, but it appears to behave consistently.

I'm not sure if my use case matches the original intent for the existing external storage constructor.  But it sounds like at least jukkasi and I intended for the last state to be loaded first by the constructor and not necessarily save again right out of the gates.
